pkg/engine/executor/pipeline.go

```diff
@@ -222,10 +222,13 @@ func (p *prefetchWrapper) Value() (arrow.Record, error) {
 
 // Close implements [Pipeline].
 func (p *prefetchWrapper) Close() {
-	// Cancel internal context so the goroutine can exit
-	p.cancel(errors.New("pipeline is closed"))
-	// Clear already pre-fetched, but unused items from channel
-	for range p.ch { // nolint:revive
+	// NOTE(rfratto): We don't need to drain p.ch because all writes to p.ch are
```

Comment:
nit: there is a potential memory leak if context cancels while waiting to send a retained record
https://github.com/grafana/loki/blob/9ba32d1815d28902ff716490926c38e666f03215/pkg/engine/executor/pipeline.go#L192